### PR TITLE
Skip non-localized error message tests on non-US locals

### PR DESCRIPTION
**Pull request by TyOverby**

### CODE REVIEW

**Comment by TyOverby** [comment]
@jaredpar 

I'm not certain that this is correct yet (I didn't even run tests on my dev box (I need to catch the bus)), I'm going to give it another look over when I get home, but this *should* work.

File: src/ExpressionEvaluator/Core/Test/ResultProvider/ResultProviderTestBase.cs [hunk 1]
```diff
@@ -307,6 +308,7 @@ internal DkmEvaluationResult[] GetItems(DkmEvaluationResultEnumContext enumConte
                 null);
         }
 
+        [ConditionalFact(typeof(IsEnglishLocal))]
```
Comment (jaredpar) [inline src/ExpressionEvaluator/Core/Test/ResultProvider/ResultProviderTestBase.cs]: Why did this get added?  This wasn't previously a test method.

File: src/Test/Utilities/Shared/Assert/ConditionalFactAttribute.cs [hunk 2]
```diff
@@ -8,12 +8,15 @@ namespace Roslyn.Test.Utilities
...
     public class ConditionalFactAttribute : FactAttribute
     {
-        public ConditionalFactAttribute(Type skipCondition)
+        public ConditionalFactAttribute(params Type[] skipConditions)
         {
-            ExecutionCondition condition = (ExecutionCondition)Activator.CreateInstance(skipCondition);
-            if (condition.ShouldSkip)
-            {
-                Skip = condition.SkipReason;
+            foreach (var skipCondition in skipConditions) {
```
Comment (jaredpar) [inline src/Test/Utilities/Shared/Assert/ConditionalFactAttribute.cs]: Braces on new lines

File: src/Test/Utilities/Shared/Assert/ConditionalFactAttribute.cs [hunk 3]
```diff
@@ -37,4 +40,12 @@ public class HasShiftJisDefaultEncoding : ExecutionCondition
 
         public override string SkipReason => "OS default codepage is not Shift-JIS (932).";
     }
+
+    public class IsEnglishLocal : ExecutionCondition
+    {
+        public override bool ShouldSkip => 
+                System.Globalization.CultureInfo.CurrentCulture == new System.Globalization.CultureInfo("en-US");
```
Comment (jaredpar) [inline src/Test/Utilities/Shared/Assert/ConditionalFactAttribute.cs]: Shouldn't this be != instead of ==?

Comment (TyOverby) [inline src/Test/Utilities/Shared/Assert/ConditionalFactAttribute.cs]: Clearly when I wrote "I'm not certain this is correct yet" I should have said "I hurried to get out the door to get to my bus, so I haven't re-read any of the code that I wrote".

**Comment by jaredpar** [comment]
:+1:

**Comment by TyOverby** [comment]
@jaredpar, how should I go about testing this?

File: src/Compilers/CSharp/Test/Emit/Attributes/InternalsVisibleToAndStrongNameTests.cs [hunk 4]
```diff
@@ -1739,7 +1739,7 @@ public class C
             }
         }
 
-        [Fact]
+        [ConditionalFact(typeof(IsEnglishLocal))]
```
Comment (TyOverby) [inline src/Compilers/CSharp/Test/Emit/Attributes/InternalsVisibleToAndStrongNameTests.cs]: This one shouldn't need to be conditional.  I don't see any hard-coded strings that would affect it.

File: src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenShortCircuitOperatorTests.cs [hunk 5]
```diff
@@ -5865,7 +5865,7 @@ .maxstack  2
         }
 
         [WorkItem(825, "https://github.com/dotnet/roslyn/issues/825")]
-        [Fact]
+        [ConditionalFact(typeof(IsEnglishLocal))]
```
Comment (TyOverby) [inline src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenShortCircuitOperatorTests.cs]: This test should probably be rewritten to not need the conditional fact.  Is there any way to get the localization info from the exception?

https://github.com/dotnet/roslyn/issues/5662

**Comment by TyOverby** [comment]
test this please

**Pull request merged**